src/main/java/mat/client/shared/MeasureLibraryResultTable.java

```diff
@@ -143,101 +149,139 @@ public void onBrowserEvent(Context context, Element elem, ManageMeasureSearchMod
+					new ClickableSafeHtmlCell()) {
+				@Override
+				public SafeHtml getValue(Result object) {
+					return getShareColumnToolTip(object);
+				}
+			};
+			shareColumn.setFieldUpdater(new FieldUpdater<ManageMeasureSearchModel.Result, SafeHtml>() {
+				@Override
+				public void update(int index, ManageMeasureSearchModel.Result object, SafeHtml value) {
+					if (object.isSharable())
+						observer.onShareClicked(object);
+				}
+			});
+			table.addColumn(shareColumn, SafeHtmlUtils.fromSafeConstant("<span title='Share'>" + "Share" + "</span>"));
 
-		// Clone
-		Column<ManageMeasureSearchModel.Result, SafeHtml> cloneColumn = new Column<ManageMeasureSearchModel.Result, SafeHtml>(
-				new ClickableSafeHtmlCell()) {
-			@Override
-			public SafeHtml getValue(Result object) {
-				return getCloneColumnHTML(object);
-			}
-		};
-		cloneColumn.setFieldUpdater(new FieldUpdater<ManageMeasureSearchModel.Result, SafeHtml>() {
-			@Override
-			public void update(int index, ManageMeasureSearchModel.Result object, SafeHtml value) {
-				if (object.isClonable())
-					Mat.showLoadingMessage();
-					observer.onCloneClicked(object);
-			}
-		});
-		table.addColumn(cloneColumn, SafeHtmlUtils.fromSafeConstant("<span title='Clone'>" + "Clone" + "</span>"));
+			// Clone
+			Column<ManageMeasureSearchModel.Result, SafeHtml> cloneColumn = new Column<ManageMeasureSearchModel.Result, SafeHtml>(
+					new ClickableSafeHtmlCell()) {
+				@Override
+				public SafeHtml getValue(Result object) {
+					return getCloneColumnHTML(object);
+				}
+			};
+			cloneColumn.setFieldUpdater(new FieldUpdater<ManageMeasureSearchModel.Result, SafeHtml>() {
+				@Override
+				public void update(int index, ManageMeasureSearchModel.Result object, SafeHtml value) {
+					if (object.isClonable())
+						Mat.showLoadingMessage();
+						observer.onCloneClicked(object);
+				}
+			});
+			table.addColumn(cloneColumn, SafeHtmlUtils.fromSafeConstant("<span title='Clone'>" + "Clone" + "</span>"));
 
-		// Export Column header
+			// Export Column header
+			Header<SafeHtml> bulkExportColumnHeader = getBulkExportColumnHeader(displayBulkExport);
+			bulkExportColumnHeader.setUpdater(new ValueUpdater<SafeHtml>() {
+				@Override
+				public void update(SafeHtml value) {
+					clearBulkExportCheckBoxes();
+				}
+			});
 
-		Header<SafeHtml> bulkExportColumnHeader = getBulkExportColumnHeader(displayBulkExport);
-		bulkExportColumnHeader.setUpdater(new ValueUpdater<SafeHtml>() {
-			@Override
-			public void update(SafeHtml value) {
-				clearBulkExportCheckBoxes();
+			table.addColumn(new Column<Result, Result>(getCompositeCellForBulkExport(displayBulkExport)) {
+				@Override
+				public Result getValue(Result object) {
+					return object;
+				}
+			}, bulkExportColumnHeader);
+		}
+		// Add event handler for table
+		table.addCellPreviewHandler(event -> {
+			String eventType = event.getNativeEvent().getType();
+			EventTarget target = event.getNativeEvent().getCurrentEventTarget();
```

Comment:
@RohitKandimalla - sholuld we try the same for CQL Library grids?

Reply:
I tried using addCellPreviewHandler, but couldn’t make it work.

Reply:
I can take a look and help if we want similar logic in CQL lib page